ports/stm32f4/common-hal/busio/SPI.c

```diff
@@ -97,10 +97,12 @@ void common_hal_busio_spi_construct(busio_spi_obj_t *self,
                 if (mcu_spi_mosi_list[j].pin == mosi) {
                     //miso
                     for(uint k=0; k<miso_len;k++) {
-                        if ((mcu_spi_miso_list[k].pin == miso)
+                        if ((mcu_spi_miso_list[k].pin == miso) //everything needs the same index
                             && (mcu_spi_sck_list[i].spi_index == mcu_spi_mosi_list[j].spi_index)
                             && (mcu_spi_sck_list[i].spi_index == mcu_spi_miso_list[k].spi_index)) {
-                            //everything needs the same index
+                            //keep looking if the SPI is taken, edge case
+                            if(reserved_spi[mcu_spi_sck_list[i].spi_index-1]) continue;
```

Comment:
I don't think you need the check below anymore since self->sck won't be set. Do you want to set a bool here to check below to produce the better error message?

Reply:
well, if I set a bool, I have to then de-set it if it actually finds one after the first failure, so that's three lines of code and a new variable to replace one check. Would you still prefer that?

Reply:
You don't need to unset it, only check it if the pins are NULL. (Before the "Invalid SPI pin selection" error.)